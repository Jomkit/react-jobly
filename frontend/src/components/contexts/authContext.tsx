import { createContext } from 'react';
import { authInterface } from '../../types';

export const authContext = createContext<authInterface>({} as authInterface);