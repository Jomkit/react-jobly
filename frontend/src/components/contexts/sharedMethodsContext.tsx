import { createContext } from 'react';
import { sharedMethodsInterface } from '../../types';

export const sharedMethodsContext = createContext<sharedMethodsInterface>({} as sharedMethodsInterface);