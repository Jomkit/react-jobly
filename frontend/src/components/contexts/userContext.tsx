import { createContext } from "react";
import { userInterface } from "../../types";

export const userContext = createContext<userInterface | null>(null);