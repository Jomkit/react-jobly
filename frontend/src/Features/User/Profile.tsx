import { useContext } from "react";
import { userContext } from "../../components/contexts/userContext";
import ProfileUpdate from "./ProfileUpdate";

const Profile = () => {
  const currUser = useContext(userContext);
  if(currUser){
    return (
      <div>
        <div className="text-white">
          <h1 className="display-1">{currUser.username}</h1>
          <p>Number of applications: {currUser.applications && currUser.applications.length}</p>
        </div>
          <ProfileUpdate />
      </div>
    )
  }
}

export default Profile