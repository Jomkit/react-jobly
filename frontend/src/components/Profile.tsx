import { useContext } from "react"
import { userContext } from "./contexts/userContext"
import { Navigate } from "react-router-dom";
import ProfileUpdate from "./ProfileUpdate";

const Profile = () => {
  const currUser = useContext(userContext);
  if(!currUser) {
    console.log(currUser);
    return (<Navigate to="/unauthorized-access" />);
  }

  if(currUser){
    return (
      <div>
          <h1 className="display-1">{currUser.username}</h1>
          <ProfileUpdate />
      </div>
    )
  }
}

export default Profile