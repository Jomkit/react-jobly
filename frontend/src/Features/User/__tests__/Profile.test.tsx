import { MemoryRouter } from "react-router-dom"
import Profile from "../Profile"
import { render } from "@testing-library/react"
import { userContext } from "../../../components/contexts/userContext"
import { TESTUSER } from "../../../__tests__/_testCommon"

const TestProfile = () => {
    return (
        <MemoryRouter>
            <userContext.Provider value={TESTUSER}>
                <Profile />
            </userContext.Provider>
        </MemoryRouter>
    )
}

it('should render without crashing', () => { 
    render(<TestProfile />);
})

it('should match snapshot', () => { 
    const { asFragment } = render(<TestProfile />);
    expect(asFragment()).toMatchSnapshot();  
})