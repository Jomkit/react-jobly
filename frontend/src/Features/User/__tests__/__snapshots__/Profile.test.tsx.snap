// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="text-white"
    >
      <h1
        class="display-1"
      >
        testuser
      </h1>
      <p>
        Number of applications: 0
      </p>
    </div>
    <div>
      <h1
        class="mt-4 text-white"
      >
        Edit User Profile
      </h1>
      <form
        action="#"
        class="d-flex flex-column col col-6 m-auto p-5 pt-3 border border-3 rounded border-dark bg-white"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          class="text-input"
          disabled=""
          id="username"
          name="username"
          type="text"
          value="testuser"
        />
        <label
          for="firstName"
        >
          First Name
        </label>
        <input
          class="text-input"
          id="firstName"
          name="firstName"
          type="text"
          value="first"
        />
        <label
          for="lastName"
        >
          Last Name
        </label>
        <input
          class="text-input"
          id="lastName"
          name="lastName"
          type="text"
          value="last"
        />
        <label
          for="email"
        >
          Email
        </label>
        <input
          class="text-input"
          id="email"
          name="email"
          type="email"
          value="test@gmail.com"
        />
        <button
          class="btn btn-primary mt-2"
          type="submit"
        >
          Save Changes
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
